pgxn/remotexact/remotexact.c

```diff
@@ -146,8 +135,15 @@ rwset_add_region(int region)
 	Assert(RegionIsValid(region));
 	Assert(rwset_collection_buffer != NULL);
 
-	// Mark the xact as remote in the procarray before adding the region.
-	set_remotexact_procflags();
+	/* Mark the xact as remote in the procarray before adding the region.
+	 * We do this by setting isRemoteXact to ture. Since we are writing to
+	 * our own process, we don't need to lock the ProcArray. Also since its
+	 * a volatile variable, we know that any subsequent reader will see the
+	 * value as true. Any process running a remotexact will idenitfy this
+	 * process as executing a remotexact while checking for deadlocks, thus
+	 * detecting a potential deadlock. 
+	 */
+	MyProc->isRemoteXact = true;
```

Comment:
`rwset_add_region` is called for every transaction, including single-region transaction. because we don't know if a single-region will be upgraded to multi-region and we need to record all regions involving in a transaction; so you need to check `region_set` to make sure that there are more than 1 region involved here.

Reply:
Done.